internal/handlers/common/update.go

```diff
@@ -208,3 +272,66 @@ func extractValueFromUpdateOperator(op string, update *types.Document) (*types.D
 		return nil, NewWriteErrorMsg(ErrFailedToParse, "Modifiers operate on fields but we found another type instead")
 	}
 }
+
+// checkCurrentDateFieldExpression checks $currentDate input on correctness.
+func checkCurrentDateFieldExpression(update *types.Document) error {
+	currentDateTopField, err := update.Get("$currentDate")
+	if err != nil {
+		return nil // it is ok: key is absent
+	}
+
+	currentDateExpression, ok := currentDateTopField.(*types.Document)
+	if !ok {
+		return NewWriteErrorMsg(
+			ErrFailedToParse,
+			"Modifiers operate on fields but we found another type instead",
+		)
+	}
+
+	for _, field := range currentDateExpression.Keys() {
+		setValue := must.NotFail(currentDateExpression.Get(field))
+
+		switch setValue := setValue.(type) {
+		case bool:
+			continue
+
+		case *types.Document:
+			for _, k := range setValue.Keys() {
+				if k != "$type" {
+					return NewWriteErrorMsg(
+						ErrBadValue,
+						fmt.Sprintf("Unrecognized $currentDate option: %s", k),
+					)
+				}
+			}
+			currentDateType, err := setValue.Get("$type")
+			if err != nil { // ok, default is date
+				continue
+			}
+
+			currentDateTypeString, ok := currentDateType.(string)
+			if !ok {
+				return NewWriteErrorMsg(
+					ErrBadValue,
+					"The '$type' string field is required to be 'date' or 'timestamp'",
+				)
+			}
+			if !slices.Contains([]string{"date", "timestamp"}, currentDateTypeString) {
```

Comment:
Do we need `slices.Contains` here? Why just simple `currentDateTypeString != "date" && currentDateTypeString != "timestamp"` wouldn't be enough?

Reply:
> Do we need `slices.Contains` here? Why just simple `currentDateTypeString != "date" && currentDateTypeString != "timestamp"` wouldn't be enough?


I find the expression `currentDateTypeString != "date" && currentDateTypeString != "timestamp"` difficult to read. See, it has to be with a negation of two:

`! ( currentDateTypeString == "date" || 
    currentDateTypeString == "timestamp")`